### ISSUE TITLE
Leaks incognito data

### ISSUE THREAD

**Issue by brian6932** [machsix/Super-preloader#485]
**Describe the bug**
This has been happening for a while and it's gotten a bit annoying at this point, If allowed in incognito is enabled for ViolentMonkey, and Super Preloader is toggled on, data leaks to https://myactivity.google.com/activitycontrols/webandapp regardless of sign in state
![image](https://user-images.githubusercontent.com/18603393/104747045-06398380-571e-11eb-832f-cce48081ad43.png)

![image](https://user-images.githubusercontent.com/18603393/104747128-223d2500-571e-11eb-995b-5fb7da63a469.png)

**Desktop (please complete the following information):**
 - Browser: Chromium
 - Version: 86
 - Script Manager: ViolentMonkey 2.12.9
 - Adblocker: uBlock Origin 1.32.5b10
 - Script version: 6.15.8
 
I will submit this post on ViolentMonkey's issues page as well, since I'm surprised there isn't any protection for this at the UserScripts Manager level https://github.com/violentmonkey/violentmonkey

**Comment by machsix** [comment]
Can you check your browser console when you visit Google? There should be something like the following
![Screenshot from 2021-01-16 00-01-48](https://user-images.githubusercontent.com/28209092/104806670-1db05500-578e-11eb-9182-72bfd0593943.png)

Please provide the content of `[Super-preloader] GM_xmlhttpRequest `

I don't understand why this could happen because I only make the script add `document.cookie` explicitly to the request header  if you are running certain old versions of the browser or script manager, which is apparently not your case
https://github.com/machsix/Super-preloader/blob/master/src/index.js#L39-L56
https://github.com/machsix/Super-preloader/blob/master/src/utils/got.js#L144-L156

The script manager will handle how credential information is sent. I only add a referrer to the header and add `overrideMimeType: "text/html; charset=UTF-8" ` I also set `withCredentials = true` but this is the default behavior of VM.
https://github.com/violentmonkey/violentmonkey/blob/e03e28b29a63d6790a24f5924f8419cf89e5b620/src/injected/web/requests.js#L139-L143

So in summary, I didn't make special changes to how XHR is sent. It might be the issue of the script manager or the **browser**.

**I highlighted the browser here because when I tested with Firefox 84.0.0, VM 2.12.8, I didn't see data leakage.**

**Comment by brian6932** [comment]
It looks like this in incognito:
![image](https://user-images.githubusercontent.com/18603393/104818172-dfe61780-57f3-11eb-82c8-06c4e0fedb58.png)
And it looks like this without incognito:
![image](https://user-images.githubusercontent.com/18603393/104818187-0b690200-57f4-11eb-9636-b8fc8967a7a6.png)

**Comment by brian6932** [comment]
I forgot to close this xD
